File: Classes/UI/listViewSorter.vb
'nexIRC 3.0.31
'05-30-2016 - guideX
Option Explicit On
Option Strict On

Public Class ListViewSorter
    Implements IComparer
    Private col As Integer
    Private order As SortOrder

    Public Sub New()
        col = 0
        order = SortOrder.Ascending
    End Sub

    Public Sub New(column As Integer, order As SortOrder)
        col = column
        Me.order = order
    End Sub

    Public Function Compare(x As Object, y As Object) As Integer Implements System.Collections.IComparer.Compare
        Dim returnVal As Integer = -1
        returnVal = [String].Compare(CType(x, ListViewItem).SubItems(col).Text, CType(y, ListViewItem).SubItems(col).Text)
        If (order = SortOrder.Descending) Then returnVal *= -1
        Return returnVal
    End Function
End Class

File: nexIRC.Business/UI/ListView/ListViewSorter.cs
//nexIRC 3.0.31
//05-30-2016 - guideX
using System;
using System.Collections;
using System.Windows.Forms;
namespace nexIRC.Business.UI.ListView {
    /// <summary>
    /// List View Sorter
    /// </summary>
    public class ListViewSorter : IComparer {
        /// <summary>
        /// Col
        /// </summary>
        private int _col;
        /// <summary>
        /// Order
        /// </summary>
        private SortOrder _order;
        /// <summary>
        /// Entry Point
        /// </summary>
        public ListViewSorter() {
            _col = 0;
            _order = SortOrder.Ascending;
        }
        /// <summary>
        /// List View Sorter
        /// </summary>
        /// <param name="column"></param>
        /// <param name="order"></param>
        public ListViewSorter(int column, SortOrder order) {
            _col = column;
            _order = order;
        }
        /// <summary>
        /// Compare
        /// </summary>
        /// <param name="x"></param>
        /// <param name="y"></param>
        /// <returns></returns>
        public int Compare(object x, object y) {
            var returnVal = -1;
            returnVal = String.Compare(((ListViewItem)x).SubItems[_col].Text, ((ListViewItem)y).SubItems[_col].Text);
            if ((_order == SortOrder.Descending)) {
                returnVal *= -1;
            }
            return returnVal;
        }
    }
}